### PR TITLE
[TRANSFORMATIONS] Add a check to SDPAToPagedAttention to fail fast

### PR DESCRIPTION
**Pull request by CuriousPanCake**
[TRANSFORMATIONS] Add a check to SDPAToPagedAttention to fail fast

Add a check to SDPAToPagedAttention to fail fast if no ScaledDotProductAttention operation present in a graph hence no transformation will be executed.

### Tickets:
 - CVS-143067

Signed-off-by: Andrii Staikov <andrii.staikov@intel.com>

### CODE REVIEW

File: src/core/src/pass/sdpa_to_paged_attention.cpp [hunk 1]
```diff
@@ -29,6 +30,16 @@ static std::shared_ptr<v0::Parameter> setName(std::shared_ptr<v0::Parameter> nod
...
     RUN_ON_MODEL_SCOPE(SDPAToPagedAttention);
+
+    for (auto& op: model->get_ordered_ops()) {
+        if (std::dynamic_pointer_cast<ov::op::v13::ScaledDotProductAttention>(op)) {
+            break;
+        }
+
+        std::cout << "No ScaledDotProductAttention operation observed in the graph, cannot perform the SDPAToPagedAttention transformation." << std::endl;
+        return false;
+    }
```
[the remark this answers is not in the document]
Comment (CuriousPanCake) [reply src/core/src/pass/sdpa_to_paged_attention.cpp:38]: Great, thanks for the suggestion! Fixed.

File: src/core/src/pass/sdpa_to_paged_attention.cpp [hunk 2]
```diff
@@ -29,6 +30,16 @@ static std::shared_ptr<v0::Parameter> setName(std::shared_ptr<v0::Parameter> nod
 
 bool ov::pass::SDPAToPagedAttention::run_on_model(const std::shared_ptr<ov::Model>& model) {
     RUN_ON_MODEL_SCOPE(SDPAToPagedAttention);
+
+    for (auto& op: model->get_ordered_ops()) {
+        if (std::dynamic_pointer_cast<ov::op::v13::ScaledDotProductAttention>(op)) {
+            break;
+        }
+
+        std::cout << "No ScaledDotProductAttention operation observed in the graph, cannot perform the SDPAToPagedAttention transformation." << std::endl;
```
[the remark this answers is not in the document]
Comment (CuriousPanCake) [reply src/core/src/pass/sdpa_to_paged_attention.cpp]: Okay, where should we print the error to?

File: src/core/src/pass/sdpa_to_paged_attention.cpp [hunk 3]
```diff
@@ -30,13 +30,10 @@ static std::shared_ptr<v0::Parameter> setName(std::shared_ptr<v0::Parameter> nod
...
     RUN_ON_MODEL_SCOPE(SDPAToPagedAttention);
-
-    for (auto& op: model->get_ordered_ops()) {
-        if (std::dynamic_pointer_cast<ov::op::v13::ScaledDotProductAttention>(op)) {
-            break;
-        }
-
-        std::cout << "No ScaledDotProductAttention operation observed in the graph, cannot perform the SDPAToPagedAttention transformation." << std::endl;
+    if (!ov::op::util::has_op_with_type<ov::op::v13::ScaledDotProductAttention>(model)) {
+        std::cout << "No ScaledDotProductAttention operation observed in the graph, cannot perform the "
```
Comment (itikhono) [inline src/core/src/pass/sdpa_to_paged_attention.cpp:34]: Did we agree to add some asserts to these Transformations?
should it be also the assert?

File: src/common/transformations/tests/SDPAToPagedAttention_test.cpp [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    const auto p1 = std::make_shared<op::v0::Parameter>(element::f32, Shape{1, 32, 32});
+    const auto add = std::make_shared<op::v1::Add>(p0, p1);
+    const auto result = std::make_shared<op::v0::Result>(add);
+
+    auto model = std::make_shared<Model>(ResultVector{result}, ParameterVector{p0, p1});
+
+    ov::pass::Manager manager;
+    manager.register_pass<pass::SDPAToPagedAttention>();
+    ASSERT_FALSE(manager.run_passes(model));
+}
```
Comment (CuriousPanCake) [inline src/common/transformations/tests/SDPAToPagedAttention_test.cpp:27]: I've added a negative test that passes if no SDPA present. However, writing a test cast where PagedAttention would be constructed is a bit more complicated as I don't really know all the requirements for a sample model such that a PagedAttention would be constructed. What do you think? I believe we may skip it for now.

**Review by itikhono — APPROVED** [review]

File: src/common/transformations/tests/SDPAToPagedAttention_test.cpp (hunk 4 above)
Comment (slyalin) [reply src/common/transformations/tests/SDPAToPagedAttention_test.cpp]: It is enough for now. The cases when PA is constructed are your other tests for PA transformation.

**Review by slyalin — APPROVED** [review]

**Comment by CuriousPanCake** [comment]
The Windows pipeline build fails due to the compilation issue in the CPU plugin.

[3755/8199] Building CXX object src\plugins\intel_cpu\CMakeFiles\openvino_intel_cpu_plugin.dir\Release\src\node.cpp.obj
FAILED: src/plugins/intel_cpu/CMakeFiles/openvino_intel_cpu_plugin.dir/Release/src/node.cpp.obj 
ccache C:\BuildTools\VC\Tools\MSVC\14.40.33807\bin\Hostx64\x64\cl.exe  /nologo /TP -DENABLE_PROFILING_ALL -DENABLE_PROFILING_FIRST_INFERENCE -DENABLE_PROFILING_ITT -DIMPLEMENT_OPENVINO_RUNTIME_PLUGIN -DIN_OV_COMPONENT -DOPENVINO_STATIC_LIBRARY -DOV_BUILD_POSTFIX=\"\" -DOV_CPU_WITH_DNNL -DOV_CPU_WITH_MLAS -DOV_CREATE_EXTENSION=create_extensions_CPU -DOV_CREATE_PLUGIN=create_plugin_engine_CPU -DOV_NATIVE_PARENT_PROJECT_ROOT_DIR=\"openvino\" -DOV_THREAD=OV_THREAD_TBB -DSELECTIVE_BUILD_ANALYZER -DTBB_PREVIEW_WAITING_FOR_WORKERS=1 -D__TBB_NO_IMPLICIT_LINKAGE=1 -DCMAKE_INTDIR=\"Release\" -IC:\actions-runner\_work\openvino\openvino\openvino\src\plugins\intel_cpu\src -IC:\actions-runner\_work\openvino\openvino\openvino\src\core\reference\include -IC:\actions-runner\_work\openvino\openvino\openvino\src\core\dev_api -IC:\actions-runner\_work\openvino\openvino\openvino\src\core\include -IC:\actions-runner\_work\openvino\openvino\openvino\src\frontends\common\include -IC:\actions-runner\_work\openvino\openvino\openvino\src\inference\include -IC:\actions-runner\_work\openvino\openvino\openvino\src\inference\dev_api -IC:\actions-runner\_work\openvino\openvino\openvino\src\common\transformations\include -IC:\actions-runner\_work\openvino\openvino\openvino\src\common\low_precision_transformations\include -IC:\actions-runner\_work\openvino\openvino\openvino\src\common\itt\include -IC:\actions-runner\_work\openvino\openvino\openvino\src\common\util\include -IC:\actions-runner\_work\openvino\openvino\openvino\thirdparty\pugixml\src -IC:\actions-runner\_work\openvino\openvino\openvino\thirdparty\ittapi\ittapi\src\ittnotify -IC:\actions-runner\_work\openvino\openvino\openvino\thirdparty\ittapi\ittapi\include -IC:\actions-runner\_work\openvino\openvino\openvino\src\common\conditional_compilation\include -IC:\actions-runner\_work\openvino\openvino\openvino\src\core\shape_inference\include -IC:\actions-runner\_work\openvino\openvino\openvino\src\common\snippets\include -external:IC:\actions-runner\_work\openvino\openvino\openvino\src\plugins\intel_cpu\thirdparty\mlas\inc -external:IC:\actions-runner\_work\openvino\openvino\openvino\src\plugins\intel_cpu\thirdparty\mlas\lib -external:IC:\actions-runner\_work\openvino\openvino\openvino_build\src\plugins\intel_cpu\thirdparty\onednn\include -external:IC:\actions-runner\_work\openvino\openvino\openvino\src\plugins\intel_cpu\thirdparty\onednn\include -external:IC:\actions-runner\_work\openvino\openvino\openvino\src\plugins\intel_cpu\thirdparty\onednn\src -external:IC:\actions-runner\_work\openvino\openvino\openvino\src\plugins\intel_cpu\thirdparty\onednn\src\..\include -external:IC:\actions-runner\_work\openvino\openvino\openvino\temp\tbb\include -external:W0 /DWIN32 /D_WINDOWS /GR /EHsc /D_CRT_SECURE_NO_WARNINGS /D_SCL_SECURE_NO_WARNINGS /EHsc /Gy /W3 /bigobj /MP /wd4251 /wd4275 /wd4996 /wd4018 /wd4267 /wd4244 /wd4334 /wd4146 -DNOMINMAX /O2 /Ob2 /DNDEBUG  /sdl /guard:cf /wd4996 -std:c++20 -MD -WX /showIncludes /Fosrc\plugins\intel_cpu\CMakeFiles\openvino_intel_cpu_plugin.dir\Release\src\node.cpp.obj /FdC:\actions-runner\_work\openvino\openvino\openvino\bin\intel64\Release\openvino_intel_cpu_plugin.pdb /FS -c C:\actions-runner\_work\openvino\openvino\openvino\src\plugins\intel_cpu\src\node.cpp
C:\actions-runner\_work\openvino\openvino\openvino\src\plugins\intel_cpu\src\node.cpp(703): fatal error C1001: Internal compiler error.

File: src/common/transformations/tests/SDPAToPagedAttention_test.cpp
Comment (ilya-lavrenov) [inline src/common/transformations/tests/SDPAToPagedAttention_test.cpp:1]: please, call files as `my_new_file` instead of `MyNewFile`
image

**Review by ilya-lavrenov — CHANGES_REQUESTED** [review]

File: src/common/transformations/tests/SDPAToPagedAttention_test.cpp
Comment (CuriousPanCake) [reply src/common/transformations/tests/SDPAToPagedAttention_test.cpp:1]: fixed.

**Review by ilya-lavrenov — APPROVED** [review]

**Pull request merged**